### PR TITLE
Add `user id (sub)` from access token to function token as `issuer`

### PR DESCRIPTION
**Pull request by leerosen-cape**

### CODE REVIEW

**Comment by gavinuhma** [comment]
Just waiting for the build to be passing and then I can review

**Comment by leerosen-cape** [comment]
@bendecoste @gavinuhma Up to date now, had to "simulate" a `.config/auth` file to get the token issuer stuff to work correctly. I also improved the token tests to verify the JWT that it outputs.

**Review by bendecoste — CHANGES_REQUESTED** [review]

File: cmd/cape/cmd/token_test.go
Comment (leerosen-cape) [reply cmd/cape/cmd/token_test.go:61]: :+1:  I was going to ask you about this

File: cmd/cape/cmd/token.go
Comment (bendecoste) [inline cmd/cape/cmd/token.go:58]:
```suggestion
	// Use the AccessToken sub (user id) as the issuer for the function token.
	// The issuer is used to determine which KMS key to use inside the enclave.
```

File: cmd/cape/cmd/login.go
Comment (leerosen-cape) [inline cmd/cape/cmd/login.go:204]: @bendecoste Updated this to add JWKS validation. Also added a `GO_ENV` var to ensure we dont do this in test. Any other ideas or does this seem like an OK pattern here

**Review by justin1121 — APPROVED** [review]
LGTM. Using an environment variable to specify the environment (test or not) is not really something we've done much in the past so might want some others input. I'm fine with it for now though.

**Comment by leerosen-cape** [comment]
> LGTM. Using an environment variable to specify the environment (test or not) is not really something we've done much in the past so might want some others input. I'm fine with it for now though.

The interesting thing is its a Viper variable, so I felt it was more protected as it cant be set externally. Or would it map to an env var automagically?

**Comment by justin1121** [comment]
So if you run `go test` it automatically gets set to `test`?

**Review by justin1121 — APPROVED** [review]

File: cmd/cape/cmd/root.go
Comment (annieltan) [reply cmd/cape/cmd/root.go:140]: yeah, what is the rationale for changing this to login? the work I was doing the last few weeks was to have one uniform URL (i.e. routing https://app.capeprivacy.com/{request_uri} to the correct URLs like login.capeprivacy.com & enclave.capeprivacy.com)

Comment (leerosen-cape) [reply cmd/cape/cmd/root.go:140]: @annieltan had no idea about that work :scream:   Sorry

How do I make the static route `/.well-known/jwks.json` for `app.capeprivacy.com`  resolve to the auth0 host

**Comment by leerosen-cape** [comment]
@annieltan See https://github.com/capeprivacy/harness/pull/165

**Comment by leerosen-cape** [comment]
@justin1121 @annieltan @bendecoste thanks for FB up to date now

**Review by bendecoste — COMMENTED** [review]
lgtm overall ... will let the code owners give the official review ;)

**Review by justin1121 — DISMISSED** [review]

**Comment by justin1121** [comment]
@bendecoste I think you have to approve since you requested changes.

File: cmd/cape/cmd/token_test.go
Comment (leerosen-cape) [reply cmd/cape/cmd/token_test.go]: @bendecoste nice! updated

**Review by justin1121 — DISMISSED** [review]

File: cmd/cape/cmd/token_test.go
Comment (bendecoste) [inline cmd/cape/cmd/token_test.go:29]: You're also supposed to

```go
defer os.RemoveAll(localConfigDir)
```

Comment (leerosen-cape) [reply cmd/cape/cmd/token_test.go:29]: :cherries:

Comment (leerosen-cape) [reply cmd/cape/cmd/token_test.go:29]: @bendecoste see updated to support removeall

**Review by justin1121 — APPROVED** [review]

**Review by bendecoste — APPROVED** [review]

**Pull request merged**
